tokio/src/util/linked_list.rs

```diff
@@ -178,8 +178,12 @@ impl<L: Link> LinkedList<L, L::Target> {
     ///
     /// # Safety
     ///
-    /// The caller **must** ensure that `node` is currently contained by
-    /// `self` or not contained by any other list.
+    /// The caller **must** ensure that exactly one of the following is true:
+    /// - `node` is currently contained by `self`
+    /// - `node` is currently contained by some other list, but `node` is
```

Comment:
Given that this is a very specific requirement, maybe include a note saying that this is for the `Notify` LL and why it relies on it (in a few words). Otherwise, if I were reading this without the necessary context, I would find it very strange.